numpy/core/shape_base.py

```diff
@@ -449,13 +449,13 @@ def _shape_recursion(arrays, list_ndim, result_ndim, depth=0):
         shapes = [_shape_recursion(arr, list_ndim, result_ndim, depth+1)
                   for arr in arrays]
         concat_depth = depth + result_ndim - list_ndim
-        for shape in shapes[1:]:
-            if (shape[:concat_depth] != shapes[0][:concat_depth] or
-                    shape[concat_depth+1:] != shapes[0][concat_depth+1:]):
-                raise ValueError('Mismatched array shapes in block.')
-        shape_on_dim = sum(shape[concat_depth] for shape in shapes)
         # Take a shape, any shape
         shape = shapes[0]
+        for shape in shapes[1:]:
```

Comment:
This overwrites the `shape` on the previous line

Reply:
the only time i didn't run pytest....